src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -934,12 +933,23 @@ private NamedTypeSymbol BindNativeIntegerSymbolIfAny(IdentifierNameSyntax node,
                     return null;
             }
 
-            if (IsInsideNameof && node.Parent is ArgumentSyntax &&
-                node.Parent?.Parent?.Parent is InvocationExpressionSyntax invocation &&
-                (invocation.Expression as IdentifierNameSyntax)?.Identifier.ContextualKind() == SyntaxKind.NameOfKeyword)
+            if (node.Parent is { } parent)
             {
-                // Don't bind nameof(nint) or nameof(nuint) so that ERR_NameNotInContext is reported.
-                return null;
+                switch (parent.Kind())
+                {
+                    case SyntaxKind.Attribute:
```

Comment:
>SyntaxKind.Attribute [](start = 25, length = 20)

Consider strengthening this check against any possible future changes to the shape of the AttributeSyntax by checking that the ```node``` is at the ```Name``` position.